src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -629,7 +645,83 @@ class UnregisteredClassesDuplicationChecker : StackObj {
   }
 };
 
-void SystemDictionaryShared::check_excluded_classes() {
+void SystemDictionaryShared::scan_constant_pool(InstanceKlass* k) {
+  if (CDSConfig::is_dumping_invokedynamic()) {
+    k->constants()->find_archivable_hidden_classes();
+  }
+}
+
+bool SystemDictionaryShared::should_hidden_class_be_archived(InstanceKlass* k) {
+  assert(k->is_hidden(), "sanity");
+  if (is_registered_lambda_proxy_class(k)) {
+    return true;
+  }
+
+  if (CDSConfig::is_dumping_invokedynamic()) {
+    if (HeapShared::is_archivable_hidden_klass(k)) {
+      return true;
+    }
+
+    DumpTimeClassInfo* info = _dumptime_table->get(k);
+    if (info != nullptr && info->is_required()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// Returns true if the class should be excluded. This can be called before
+// SystemDictionaryShared::check_excluded_classes().
```

Comment:
There are a couple of references to check_excluded_classes() in the comments but this function no longer exists. Can you please update the comment accordingly.

Reply:
I replaced with the new name of the function `find_all_archivable_classes()`.